model/task_lifecycle.go

```diff
@@ -499,6 +501,62 @@ func MarkEnd(t *task.Task, caller string, finishTime time.Time, detail *apimodel
 	return nil
 }
 
+func TryDequeueAndAbortCommitQueueVersion(versionId, projectId string, requester string) error {
+	p, err := patch.FindOne(patch.ByVersion(versionId))
+	if err != nil {
+		return errors.Wrapf(err, "error finding patch")
+	}
+	if p == nil {
+		return errors.Errorf("No patch for task")
+	}
+
+	if p.Alias != evergreen.CommitQueueAlias {
+		return nil
+	}
+
+	// if task is part of a commit queue, dequeue and abort version
+	cq, err := commitqueue.FindOneId(projectId)
+	if err != nil {
+		return errors.Wrapf(err, "can't get commit queue for id '%s'", projectId)
+	}
+	issue := p.Id.Hex()
+	if p.IsGithubPRPatch() {
+		issue = strconv.Itoa(p.GithubPatchData.PRNumber)
+	}
+	removed, err := cq.Remove(issue)
+	if err != nil {
+		return errors.Wrapf(err, "can't remove item '%s' from queue '%s'", versionId, projectId)
+	}
+	if !removed {
+		return nil
+	}
+	if p.IsGithubPRPatch() {
+		env := evergreen.GetEnvironment()
+		sender, err := env.GetSender(evergreen.SenderGithubStatus)
+		if err != nil {
+			grip.Debug(message.WrapError(err, message.Fields{
+				"message":      "error getting sender for github status",
+				"patch_id":     p.Id,
+				"project":      p.Project,
+				"pull_request": issue,
+			}))
+		} else {
+			status := message.GithubStatus{
+				Context:     commitqueue.Context,
+				Description: "removed from queue because task failed",
+				State:       message.GithubStateFailure,
+				Ref:         p.GithubPatchData.HeadHash,
+				Repo:        p.GithubPatchData.HeadRepo,
+				Owner:       p.GithubPatchData.Author,
```

Comment:
I think the status will only make it to the PR with the owner/repo of the base repo (like for evergreen, the owner would be `evergreen-ci` instead of `ablack12`)